### PR TITLE
chore: make module resource-service

### PR DESCRIPTION
**Pull request by astsiapanay**

### CODE REVIEW

File: resource-service/build.gradle [hunk 1]
```diff
@@ -0,0 +1,57 @@
+plugins {
```
Comment (artsiomkorzun) [inline resource-service/build.gradle]: It is mostly same as server build.gradle. Could you move the similar parts to build.gradle at the root  project directory?

File: resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java [hunk 2]
```diff
@@ -39,7 +38,7 @@ public Subscription subscribe(Collection<ResourceDescriptor> resources, Consumer
             String url = resource.getUrl();
             urlToSubscriptions.compute(url, (key, subs) -> {
                 if (subs == null) {
-                    subs = new ConcurrentHashSet<>();
+                    subs = ConcurrentHashMap.newKeySet();
```
Comment (artsiomkorzun) [inline resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java]: why?

File: server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java [hunk 3]
```diff
@@ -1,6 +1,6 @@
 package com.epam.aidial.core.server.data;
 
-import com.epam.aidial.core.server.resource.ResourceType;
+import com.epam.aidial.core.resourceservice.resource.ResourceType;
```
Comment (artsiomkorzun) [inline server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java]: A weird package name? com.epam.aidial.core.resource.service.ResourceType?

File: resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java (hunk 2 above)
Comment (astsiapanay) [reply resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java]: we want to remove vertx dep from the module

File: server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java (hunk 3 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java]: right the package name `resourceservice` is a bit weird but
if you suggest `resource.service`
The rest packages will be `resource.service.resource` or `resource.service.service`

File: resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java (hunk 2 above)
Comment (artsiomkorzun) [reply resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java]: It comes from java.util.*

File: resource-service/build.gradle (hunk 1 above)
Comment (astsiapanay) [reply resource-service/build.gradle]: fixed

File: server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java (hunk 3 above)
Comment (astsiapanay) [reply server/src/main/java/com/epam/aidial/core/server/data/ResourceTypes.java]: let's rename to `storage`

File: resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java (hunk 2 above)
Comment (artsiomkorzun) [reply resource-service/src/main/java/com/epam/aidial/core/resourceservice/service/ResourceTopic.java]: Sorry, it comes from Vertx

**Review by artsiomkorzun — APPROVED** [review]

**Comment by astsiapanay** [comment]
/deploy-review
>Environment URL: https://chat-ai-dial-core-pr-548.nightly-test.deltixhub.io
>E2E tests status: success

**Pull request merged**